src/Mod/Part/Gui/ViewProviderGridExtension.cpp

```diff
@@ -112,6 +114,8 @@ class GridExtensionP : public ParameterGrp::ObserverType {
     bool enabled = false;
     double computedGridValue = 10;
 
+    bool isTooManySegmentsNotified = false;
```

Comment:
Couldn't that be just a static local?

Reply:
It could. 

Two caveats:
- the lifetime would be the program lifetime since the first execution of the function (even if the instance of the class is destroyed).
- it would be shared by all instances of this class. This means that changing its value in one instance affects any other instance (when calling that function). Here it may not be a problem, but still something to keep in mind.

I am unsure we should extend too much the use of statics. Where performance is at stake, we should definitely think about it, for example in the case of resources that are expensive to create and impact performance. 

I think I need quite a time of reflection and reading about what is the place of statics (taking into account its advantages and disadvantages).

Reply:
Oh, maybe I was unclear. I talk about a static variable scoped inside GridExtensionP::createGridPart function.

Reply:
https://stackoverflow.com/questions/6223355/static-variables-in-member-functions

Reply:
Oh. So I was clear. :) Just ignorant :stuck_out_tongue_closed_eyes: Thanks